config/v1/types_node.go

```diff
@@ -46,6 +46,14 @@ type NodeSpec struct {
 	// the status and corresponding reaction of the cluster
 	// +optional
 	WorkerLatencyProfile WorkerLatencyProfileType `json:"workerLatencyProfile,omitempty"`
+
+	// MinimumKubeletVersion is the lowest version of a kubelet that can meaningfully join the cluster.
+	// Specifically, the apiserver will deny (most) authorization requests of kubelets that are older
+	// than the specified version.
+	// +kubebuilder:validation:Pattern=`^[0-9]*\.[0-9]*\.[0-9]*$`
+	// +openshift:enable:FeatureGate=MinimumKubeletVersion
+	// +optional
+	MinimumKubeletVersion string `json:"minimumKubeletVersion,omitempty"`
```

Comment:
do we really want to omitempty or keep the empty value in favour of discoverability?

Reply:
I think so, mostly because the other fields in this object also have it

Reply:
cc @JoelSpeed can we define a criteria to be consistent here? Why would we add it here and decide not to here https://github.com/openshift/api/pull/2070?

Reply:
It may not be immediately obvious, but we touch on this in [our API conventions](https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#configuration-vs-workload-apis).

This is a configuration API IMO and therefore should not have the `omitempty` tag. The section on pointers talks about this a little